spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/cache/GeodeCacheConfiguration.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2011-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.cache;
+
+import java.util.HashSet;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.GemFireCache;
+import org.apache.geode.cache.client.ClientCache;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.gemfire.CacheFactoryBean;
+import org.springframework.data.gemfire.client.ClientCacheFactoryBean;
+import org.springframework.data.gemfire.support.GemfireCacheManager;
+
+/**
+ * GemFire cache configuration.
+ *
+ * @author John Blum
+ * @see org.apache.geode.cache.Cache
+ * @see org.apache.geode.cache.GemFireCache
+ * @see org.apache.geode.cache.client.ClientCache
+ * @see org.springframework.context.annotation.Bean
+ * @see org.springframework.context.annotation.Configuration
+ * @see org.springframework.data.gemfire.support.GemfireCacheManager
+ * @since 1.5.0
+ */
+@Configuration
+@ConditionalOnBean({ CacheFactoryBean.class, Cache.class,
```

Comment:
@snicoll - Alright, I see what you are saying now, but, as an FYI, I think you need to be more careful in what you mean.  It is not "GemfireCache", it is "[GemFireCache](http://geode.incubator.apache.org/releases/latest/javadoc/org/apache/geode/cache/GemFireCache.html)"

Unfortunately, there are a bunch of class names in SDG that are inappropriately named (e.g. [GemfireCache](http://docs.spring.io/spring-data-gemfire/docs/current/api/org/springframework/data/gemfire/support/GemfireCache.html)).  It has been that way since the project inception and would very disrupting to users to change the class names now.  My plan is to introduce new configuration models that will allow them to rely less on the SDG API (and naming conventions) and more on abstractions I introduce by way of Annotations, that will allow me to clean up the prior mess.

Regarding the recommendations on `@ConditionalOnBean` bean class references... the SDG `FactoryBeans` return very specific types (e.g. `Cache` and `ClientCache`).  In earlier versions of the Spring Framework (possibly 4.1 or earlier) you could not refer to and inject a GemFire/Geode cache instance generically (i.e. as in `GemFireCache`); it simply could not resolve the bean type.  In fact, I think this problem was specific to the `ClientCache` interface in particular.  Oddly enough, the actual implementing class in GemFire/Geode is `GemFireCacheImpl`, which implements both `Cache` and `ClientCache`.

Then, something changed in latter versions of core _Spring Framework_ where this started working.  I suppose I can reconsider this now since Spring Boot 1.5 is [based on](https://github.com/spring-projects/spring-boot/blob/1.5.x/spring-boot-dependencies/pom.xml#L149) core _Spring Framework_ **4.3**.